build/targets/standard-lifecycle.targets

```diff
@@ -1,57 +1,153 @@
 
-  <!-- default targets -->
   <Target Name="Build" DependsOnTargets="Clean;Restore;Compile;Package;Test;Verify" />
 
-  <Target Name="Clean">
+<!--
+###################################################################
+Convenience targets.
+
+These are not part of the standard lifecycle but are useful
+###################################################################
+-->
+
+  <Target Name="Rebuild" DependsOnTargets="RebuildSolutions" />
+
+  <!-- TODO implement. See https://github.com/aspnet/BuildTools/issues/162 -->
+  <Target Name="Resx" />
+
+<!--
+###################################################################
+Individual build steps.
+
+Repos that customize the build should not rely on these existing
+as they are subject to change.
+###################################################################
+-->
+
+  <Target Name="CleanArtifacts">
+    <!-- intentionally BuildDir, not ArtifactDir -->
     <RemoveDir Directories="$(BuildDir)" />
+  </Target>
+
+<!--
+###################################################################
+Targets: {Target}Solutions
+
+Items: Solutions
+
+Executes /t:{Target} on all solutions
+###################################################################
+-->
+  <Target Name="CleanSolutions">
     <MSBuild Targets="Clean"
-              Projects="@(Solutions)"
-              Properties="Configuration=$(Configuration)" />
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
   </Target>
 
-  <Target Name="Restore">
+  <Target Name="RestoreSolutions">
     <MSBuild Targets="Restore"
-             Projects="@(Solutions)"
-             Properties="Configuration=$(Configuration)" />
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
   </Target>
 
-  <Target Name="Compile">
+  <Target Name="BuildSolutions">
     <MSBuild Targets="Build"
-             Projects="@(Solutions)"
-             Properties="Configuration=$(Configuration)" />
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
   </Target>
 
-  <Target Name="Package">
-    <!-- TODO consider using GeneratePackageOnBuild or calling Pack on the solution. Requires setting IsPackable=false on test and samples -->
+  <Target Name="RebuildSolutions">
+    <MSBuild Targets="Rebuild"
+      Projects="@(Solutions)"
+      Properties="Configuration=$(Configuration)" />
+  </Target>
+
+<!--
+###################################################################
+Target: PackageProjects
+
+Executes /t:Pack on all projects matching src/*/*.csproj.
+###################################################################
+-->
+  <Target Name="PackageProjects">
+    <!-- TODO consider using GeneratePackageOnBuild or calling Pack on the solution.
+        This would require setting IsPackable=false and samples and possible on tests (see https://github.com/NuGet/NuGet.Client/pull/1170) -->
     <ItemGroup>
       <ProjectsToPack Include="$(RepositoryRoot)src\*\*.csproj" />
```

Comment:
`ProjectsToPack Condition="'@(ProjectsToPack)'==''"`

Assuming 1170 doesn't land, it would be nice to have a way to customize what gets packed from `repo.props`.